compiler/qsc/src/interpret/stateless.rs

```diff
@@ -163,9 +163,29 @@ impl Interpreter {
             state: State::new(self.package),
         }
     }
+
+    #[must_use]
+    pub fn new_eval_context_from_sim(
+        &self,
+        sim: impl Backend<ResultType = impl Into<val::Result>>,
```

Comment:
Do you really need two methods? Passing in `SparseSim::new()` for the original callers of `new_eval_context` seems unobtrusive.

Reply:
If I do that, then I'll need to export `SparseSim` from the `qsc` module, which today it is not. Some callers that create an eval context are outside of the compiler, like `check_solution` in katas/src/lib.rs. It seems reasonable to keep this API that is called in most places and just have the more parametrized one for the single case (so far) of QIR codegen.

Reply:
> If I do that, then I'll need to export `SparseSim` from the `qsc` module, which today it is not.

That makes the current situation even worse, because `new_eval_context` returns `EvalContext<SimContext>`. If `SimContext` is not re-exported, the return type is unnamable without depending on qsc_eval.